Plugins/Triage/Retrieve-Items.ps1

```diff
@@ -179,42 +227,134 @@ process{
         } 
 
-        Invoke-Command -Session $Session -ScriptBlock {
+        $Locked = Invoke-Command -Session $Session -ScriptBlock {$IsLocked}
+        #Process the $item based on IsLocked value
+
+        if (!$Locked){
+
+            Invoke-Command -Session $Session -ScriptBlock {
 
             #Create archive of Item and MetaData (separately)
             $ArchivePath = "C:\ProgramData\{0}.zip" -f (Split-Path $($args[1]) -Leaf)
             $Command_Compress = "C:\ProgramData\{0} a -pinfected -tzip {1} {2} {3}" -f ($($args[0]), $ArchivePath, $ExportPath, ($($args[1])))
 
             Invoke-Expression -Command $Command_Compress | Out-Null
 
-        } -ArgumentList (Split-Path $Installexe -Leaf), $Item
+            } -ArgumentList (Split-Path $Installexe -Leaf), $Item
+
+            #Copy specified archive to $Output
+            $ItemPath = "C:\ProgramData\{0}.zip" -f (Split-Path $Item -Leaf)
+
+            Copy-Item -Path $ItemPath -Destination "$Output\" -FromSession $Session -Force -ErrorAction SilentlyContinue
+
+            #Remove created files on remote machine as cleanup
+
+            Invoke-Command -Session $Session -ScriptBlock {
+
+                #Remove the archive
+
+                Remove-Item -Path $ArchivePath -Force 
+                    
+                #Remove the Metadata file
+
+                Remove-Item -Path $ExportPath -Force  
+
+            }
+        }
+
+        if ($Locked){
+
+            #Deploy Velociraptor
+
+            $Test_Velo = Invoke-Command -Session $Session -ScriptBlock {Get-Item ("C:\ProgramData\{0}" -f $($args[0])) -ErrorAction SilentlyContinue} -ArgumentList (Split-Path $Velo_exe -Leaf)
 
-        #Copy specified archive to $Output
-        $ItemPath = "C:\ProgramData\{0}.zip" -f (Split-Path $Item -Leaf)
+            if (!$Test_Velo){
 
-        Copy-Item -Path $ItemPath -Destination "$Output\" -FromSession $Session -Force -ErrorAction SilentlyContinue
+                try{
 
-        #Remove created files on remote machine as cleanup
+                    Copy-Item -Path $Velo_exe -Destination "C:\ProgramData" -ToSession $Session -ErrorAction Stop 
 
-        Invoke-Command -Session $Session -ScriptBlock {
+                }catch {
 
-            #Remove the archive
+                    Throw ("Locked file detected at {0}, but could not deploy Velociraptor for retrieval. Quitting..." -f $Item)
+                }
+
+            }
+
+            #Copy $Item with Velociraptor, compress, and copy back to origin machine
+
+            #Collect $Item
+            $FinalPath = "C:\ProgramData\{1}_{0}" -f $Session.ComputerName, (Split-Path $Item -Leaf)
+
+            Invoke-Command -Session $Session -ScriptBlock {New-Item -Type Directory -Path $($args[0])} -Argumentlist $FinalPath
+
+            $ScriptBlock = $ExecutionContext.InvokeCommand.NewScriptBlock(('& {0}\{1} fs --accessor ntfs cp \\.\{2} {3}') -f ($env:ProgramData, (Split-Path -Path $Velo_exe -Leaf), $Item, $FinalPath))
```

Comment:
Clever using `$env:ProgramData`, but earlier you hard-coded `C:\ProgramData` for the `$TestVelo` and `$FinalPath` so if the values are different for whatever reason this could fail.

Also you could make `$Velo_Exe` a File object higher up the script by doing a `$Velo_Exe = Get-Item -Path $Velo_64`. That would allow you to replace the all the `(Split-Path -Leaf $Velo_Exe)` with a cleaner `$Velo_Exe.Name`. Foodz for thought :)

Reply:
that copy and paste life...

On the other point, good to keep in mind. I think i will leave as is for now.